### PR TITLE
Add Prometheus metrics for QueryCoord

### PR DESCRIPTION
**Pull request by bigsheeper**
Signed-off-by: bigsheeper <yihao.dai@zilliz.com>

issues: #15592

### CODE REVIEW

File: internal/metrics/querycoord.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		})
+
+	QueryCoordNumEntities = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryCoordRole,
+			Name:      "num_entities",
+			Help:      "Number of entities in collection.",
+		}, []string{
+			"num_entities",
```
Comment (czs007) [inline internal/metrics/querycoord.go:51]: add collection_id label to distinguish different collections.

File: internal/metrics/querycoord.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryCoordRole,
+			Name:      "num_parent_tasks",
+			Help:      "Number of parent tasks in QueryCoord.",
+		}, []string{
+			"num_parent_tasks",
+		})
+
+	QueryCoordChildTaskExecTime = prometheus.NewHistogramVec(
```
Comment (czs007) [inline internal/metrics/querycoord.go:116]: rename to QueryCoordChildTaskLatency

File: internal/metrics/querycoord.go (hunk 1 above)
Comment (bigsheeper) [reply internal/metrics/querycoord.go:51]: fixed

File: internal/metrics/querycoord.go (hunk 2 above)
Comment (bigsheeper) [reply internal/metrics/querycoord.go]: fixed

**Review by LoveEachDay — APPROVED** [review]
/lgtm

**Pull request merged**